test/python/algorithms/quantum_time_evolution/__init__.py

```diff
@@ -0,0 +1,11 @@
+# This code is part of Qiskit.
```

Comment:
This looks like the legacy code structure 🙂 In fact, I think the whole test structure still follows our old concept. Could you update the directory names to reflect to new module names?